docs/src/main/sphinx/connector/memory.rst

```diff
@@ -78,7 +63,24 @@ DROP TABLE
 
 Upon execution of a ``DROP TABLE`` operation, memory is not released
 immediately. It is instead released after the next write operation to the
-connector.
+catalog.
```

Comment:
This is the real change.

Reply:
Yes. Seems correct. I think someone like @martint or @findepi should confirm though. I assume this is the case since this is where the cleanup happens (and also because catalogs are isolated from each other AFAIK so they can't communicate) - https://github.com/trinodb/trino/blob/03d8e5abc686c5c4a9f96ca14db07e2aed880174/plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryPageSinkProvider.java#L58-L67

Reply:
This is correct.

@mosabua please split real change from refactoring into two commits

Reply:
> This is correct.

i.e. the "catalog" is correct.

Reply:
Done